semantic-rust/src/Language/Rust/Tags.hs

```diff
@@ -50,6 +59,172 @@ gtags
+-- }
+
+yieldTag :: (Has (Reader Source) sig m, Has (Writer Tags.Tags) sig m) => Text -> P.SyntaxType -> P.NodeType -> Loc -> Range -> m ()
+yieldTag name kind ty loc srcLineRange = do
+  src <- ask @Source
+  Tags.yield (Tag name kind ty loc (Tags.firstLine src srcLineRange) Nothing)
+
+
+instance ToTags Rust.ModItem where 
+  tags 
+    t@Rust.ModItem 
+      {
+        ann = loc@Loc {byteRange},
+        body,
+        name = Parse.Success (Rust.Identifier {text, ann})
+      } = do 
+        case body of 
+          Just (Parse.Success decls) -> gtags decls
+          _ -> pure ()
+        where yield name ann = yieldTag name P.FUNCTION P.DEFINITION ann byteRange >> gtags t
+  tags _ = pure ()
+
+
+instance ToTags Rust.FunctionItem where
+  tags
+    t@Rust.FunctionItem
+      { 
+        ann = loc@Loc {byteRange},
+        returnType,
+        body = Parse.Success block,
+        name,
+        parameters = Parse.Success parameters,
+        typeParameters
+      } = do
+      case returnType of
+        Just (Parse.Success type') -> gtags type'
+        _ -> pure ()
+      case typeParameters of
+        Just (Parse.Success typeParams) -> gtags typeParams
+        _ -> pure ()
+      case name of
+        EPrj (Rust.Identifier {text, ann}) -> yield text ann
+        EPrj (Rust.Metavariable {text, ann}) -> yield text ann
+        _ -> pure ()
+      gtags block
+      where
+        yield name ann = yieldTag name P.FUNCTION P.DEFINITION ann byteRange >> gtags t
+  tags _ = pure ()
+
+
+instance ToTags Rust.CallExpression where
+  tags
+    t@Rust.CallExpression 
+      {
+        ann = loc@Loc {byteRange},
+        function = Parse.Success expr,
+        arguments = Parse.Success args
+      } = gtags expr
+  tags _ = pure ()
+
+instance ToTags Rust.EnumItem where 
+  tags 
+    t@Rust.EnumItem
+      {
+        ann = loc@Loc {byteRange},
+        body = Parse.Success bod,
+        name = Parse.Success (Rust.TypeIdentifier {text, ann})
+      } = yieldTag text P.FUNCTION P.DEFINITION ann byteRange >> gtags t
+
+
+instance ToTags Rust.FieldDeclaration where
+  tags
+    t@Rust.FieldDeclaration
+      { ann = loc@Loc {byteRange},
+        name = Parse.Success (Rust.FieldIdentifier {text, ann}),
+        type' = Parse.Success type''
+      } = yieldTag text P.FUNCTION P.DEFINITION ann byteRange >> gtags t
+  tags _ = pure ()
+
+
+instance ToTags Rust.FunctionSignatureItem where
+  tags
+    t@Rust.FunctionSignatureItem
+      { ann = loc@Loc {byteRange},
+        returnType,
+        name,
+        parameters = Parse.Success params,
+        typeParameters
+      } = do
+        case returnType of 
+          Just (Parse.Success type') -> gtags type'
+          _ -> pure ()
+        case name of
+          EPrj (Rust.Identifier {text, ann}) -> yield text ann
+          EPrj (Rust.Metavariable {text, ann}) -> yield text ann
+      where
+        yield name ann = yieldTag name P.FUNCTION P.DEFINITION ann byteRange >> gtags t
```

Comment:
Hard to say whether this is the right thing without knowing what a FunctionSignatureItem constitutes.